src/Controller/HealthController.php

```diff
@@ -4,15 +4,50 @@
 
 namespace Akondas\ActuatorBundle\Controller;
 
+use Akondas\ActuatorBundle\Service\Health\HealthIndicator;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
 
 class HealthController extends AbstractController
 {
+    public const HEALTHY_UP = 'UP';
+    public const HEALTHY_DOWN = 'DOWN';
+
+    /**
+     * @var HealthIndicator[]
+     */
+    private iterable $handlers;
+
+    /**
+     * @param HealthIndicator[] $handlers
+     */
+    public function __construct(iterable $handlers)
+    {
+        $this->handlers = $handlers;
+    }
+
     public function health(): JsonResponse
     {
-        return new JsonResponse([
-            'status' => 'up',
-        ]);
+        $healthy = true;
+        $details = [];
+        foreach ($this->handlers as $handler) {
+            $handlerHealth = $handler->health();
+            $status = $handlerHealth->getStatus();
+            if ($status === false) {
+                $healthy = false;
+            }
+
+            $details[$handler->name()] = ['status' => $status ? self::HEALTHY_UP : self::HEALTHY_DOWN, 'details' => $handlerHealth->getDetails()];
+        }
+
+        $response = [
+            'status' => $healthy ? self::HEALTHY_UP : self::HEALTHY_DOWN,
+        ];
+
+        if (count($details) !== 0) {
+            $response['details'] = $details;
+        }
+
+        return new JsonResponse($response, $healthy ? 200 : 503);
```

Comment:
I would move this logic to another service, maybe `HealthIndicatorStack` or `HealthIndicators`. This service can return object that will implement `JsonSerializable` and then you can just past this to JsonResponse constructor

Reply:
done with a little caveat

because i need the result of the health indicators for the http status code, I would have either:
* Make the checks first and compare the result
* Save the status in a state member variable of the stack

the latter would mean it's easier to use but I think the state should not be in the stack, so I implemented the first variant

but i'm open for feedback

Reply:
https://github.com/akondas/symfony-actuator-bundle/pull/2/files#r876656538